dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,292 @@
+
+        if not 0 <= alpha <= 1:
+            raise ValueError(f"alpha must be between 0 and 1, given {alpha}")
+
+        if num_features <= 0:
+            raise ValueError(f"num_features must be a positive integer, given {num_features}")
+
+        cqm = dimod.ConstrainedQuadraticModel()
+        cqm.add_variables(dimod.BINARY, X.shape[1])
+
+        # add the k-hot constraint
+        cqm.add_constraint(((v, 1) for v in cqm.variables), '==' if strict else '<=', num_features)
+
+        with tempfile.TemporaryFile() as fX, tempfile.TemporaryFile() as fout:
+            # we make a copy of X because we'll be modifying it in-place within
+            # some of the functions
+            X_copy = np.memmap(fX, X.dtype, mode="w+", shape=(X.shape[0], X.shape[1] + 1))
+            X_copy[:, :-1] = X
+            X_copy[:, -1] = y
+
+            # make the matrix that will hold the correlations
+            correlations = np.memmap(
+                fout,
+                dtype=np.result_type(X, y),
+                mode="w+",
+                shape=(X_copy.shape[1], X_copy.shape[1]),
+                )
+
+            # main calculation. It modifies X in-place
+            corrcoef(X_copy, out=correlations, rowvar=False, copy=False)
+
+            # we don't care about the direction of correlation in terms of
+            # the penalty/quality
+            np.absolute(correlations, out=correlations)
+
+            # our objective
+            np.fill_diagonal(correlations, -correlations[:, -1] * alpha)
+
+            # Note: the full symmetric matrix (with both upper- and lower-diagonal
+            # entries for each correlation coefficient) is retained for consistency with
+            # the original formulation from Milne et al.
+            it = np.nditer(correlations[:-1, :-1], flags=['multi_index'], op_flags=[['readonly']])
+            cqm.set_objective((*it.multi_index, x) for x in it)
+
+        return cqm
+
+    def fit(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: typing.Optional[float] = None,
+        num_features: typing.Optional[int] = None,
+        time_limit: typing.Optional[float] = None,
+    ) -> SelectFromQuadraticModel:
+        """Select the features to keep.
+
+        Args:
+            X:
+                2D array-like of feature vectors (numerical).
+            y:
+                1D array-like of class labels (numerical).
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+                Defaults to the value provided to the constructor.
+            num_features:
+                The number of features to select.
+                Defaults to the value provided to the constructor.
```

Comment:
```suggestion
                Defaults to the value set for the constructor.
```

Because the constructor also has a default so might not be provided there by user

Reply:
I don't agree. The default is providing the input parameter to the function. And it defaults to the value from the constructor always, not just when specified by the user.

Reply:
That would be true if it said "provided from"; "provided to the constructor" connotates a user providing the constructor (not the function!) a value versus "set for" which can be a user provided value or a preconfigured value. MHO.